transaction.go

```diff
@@ -0,0 +1,53 @@
+package storm
+
+// Begin starts a new transaction.
+func (n *Node) Begin(writable bool) (*Node, error) {
```

Comment:
Ah sorry, I read this a little bit too fast. OK, then I like it a lot.

I would, however, use the Node value as receiver, to make the copying explicit in the method signature:

```go
func (n Node) Begin(writable bool) (*Node, error)
```
This also saves one code line.